js/tabRestore.js

```diff
@@ -29,8 +47,30 @@ TabHistory.prototype.restore = function () {
     return
   }
 
-  var newIndex = tabs.getIndex(tabs.getSelected()) + 1
-  var newTab = tabs.add(this.stack.pop(), newIndex)
+  var stackTop = this.stack.pop()
+  var parentTask = stackTop.task
+
+  if (tasks.get(parentTask.id) === null) {
+    // The task has since been destroyed. Create a new one.
+    parentTask.tabs = []
+    tasks.add(parentTask)
+  }
+
+  switchToTask(parentTask.id)
+  lastTab = getSelectedTask().tabs[parentTask.tabs.length - 1]
```

Comment:
BTW, `window.tabs` already stores the tabs in the current task (tabState.js, lines 175-176).